### PR TITLE
HBASE-27775 Use a separate WAL provider for hbase:replication table

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Comment by Apache9** [comment]
This is for fixing the TestMultiSlaveReplication, and also to avoid generating a lot of useless data in hbase:replication table, while nothing written to the cluster.

The current implementation may still not enough as we may still track the WAL file for hbase:replication, but anyway it will not mess up the normal WAL files.

Will dig more and report back here.

**Comment by Apache9** [comment]
OK, confirmed, there is no problem. We will add a WALActionListener in Replication for tracking the WAL files which need to be replicated, and it will only add the listener to user space WALProvider, so we are safe here.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-27109/table_based_rqs Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 28s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  compile  |   3m 38s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  checkstyle  |   0m 47s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  spotless  |   0m 51s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m  9s |  HBASE-27109/table_based_rqs passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 47s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 25s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 15s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  20m 31s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   1m  3s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m 59s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 23s |  The patch does not generate ASF License warnings.  |
|  |   |  62m  9s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5157 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 4b92f3fd3627 5.4.0-1097-aws #105~18.04.1-Ubuntu SMP Mon Feb 13 17:50:57 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27109/table_based_rqs / 49589ef285 |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 85 (vs. ulimit of 30000) |
| modules | C: hbase-replication hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/2/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27109/table_based_rqs Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   2m 58s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  compile  |   0m 44s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  shadedjars  |   4m 30s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 31s |  HBASE-27109/table_based_rqs passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   2m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 43s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 28s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 29s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 22s |  hbase-replication in the patch passed.  |
| +1 :green_heart: |  unit  | 217m  7s |  hbase-server in the patch passed.  |
|  |   | 239m 31s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5157 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 8bf5210e32d6 5.4.0-1097-aws #105~18.04.1-Ubuntu SMP Mon Feb 13 17:50:57 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27109/table_based_rqs / 49589ef285 |
| Default Java | Temurin-1.8.0_352-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/2/testReport/ |
| Max. process+thread count | 3239 (vs. ulimit of 30000) |
| modules | C: hbase-replication hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/2/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 26s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-27109/table_based_rqs Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 48s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  compile  |   2m 45s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  checkstyle  |   0m 41s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  spotless  |   0m 42s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   1m 49s |  HBASE-27109/table_based_rqs passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 32s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 40s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 40s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  14m  2s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 41s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 16s |  The patch does not generate ASF License warnings.  |
|  |   |  42m 59s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5157 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 9171c915f202 5.4.0-1097-aws #105~18.04.1-Ubuntu SMP Mon Feb 13 17:50:57 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27109/table_based_rqs / f78fe5994a |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 81 (vs. ulimit of 30000) |
| modules | C: hbase-replication hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/3/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 42s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-27109/table_based_rqs Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 46s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  compile  |   3m 21s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  checkstyle  |   0m 58s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  spotless  |   0m 58s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m 32s |  HBASE-27109/table_based_rqs passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 11s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m 43s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m 15s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m 15s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m  4s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  22m  6s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   1m  6s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   3m 10s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 23s |  The patch does not generate ASF License warnings.  |
|  |   |  64m 30s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5157 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 034e447686f5 5.4.0-1097-aws #105~18.04.1-Ubuntu SMP Mon Feb 13 17:50:57 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27109/table_based_rqs / f78fe5994a |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-replication hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/4/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 51s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  2s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27109/table_based_rqs Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 34s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  compile  |   1m  1s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  shadedjars  |   4m 46s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  HBASE-27109/table_based_rqs passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 39s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 38s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 28s |  hbase-replication in the patch passed.  |
| -1 :x: |  unit  | 235m 26s |  hbase-server in the patch failed.  |
|  |   | 261m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/4/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5157 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux ac3673b3d7e1 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27109/table_based_rqs / f78fe5994a |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/4/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/4/testReport/ |
| Max. process+thread count | 2859 (vs. ulimit of 30000) |
| modules | C: hbase-replication hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/4/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/wal/LazyInitializedWALProvider.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+          conf.get(WALFactory.WAL_PROVIDER, WALFactory.DEFAULT_WAL_PROVIDER));
+      }
+      provider = WALFactory.createProvider(clz);
+      provider.init(factory, conf, providerId, this.abortable);
+      provider.addWALActionsListener(new MetricsWAL());
+      if (this.holder.compareAndSet(null, provider)) {
+        return provider;
+      } else {
+        // someone is ahead of us, close and try again.
+        provider.close();
```
Comment (2005hithlj) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/wal/LazyInitializedWALProvider.java:81]: @Apache9 sir.
After closing here, is the provider retrieved from this.holder.get() (57th line of code)  that has already been closed？

**Comment by 2005hithlj** [comment]
@Apache9  sir.
Also, do we need to add a new UT to cover our newly added code?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/wal/LazyInitializedWALProvider.java (hunk 1 above)
Comment (Apache9) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/wal/LazyInitializedWALProvider.java:81]: The code is just a copy of the old code in WALFactory. Here we just close the loser, i.e, the one which is failed to set to the class field, so we need to close it. No one will use it.

**Comment by Apache9** [comment]
I think we could add a UT to confirm that we write the WAL for hbase:replication to a separated file.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/wal/LazyInitializedWALProvider.java (hunk 1 above)
Comment (2005hithlj) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/wal/LazyInitializedWALProvider.java:81]: ok, I get it.

**Comment by Apache9** [comment]
> I think we could add a UT to confirm that we write the WAL for hbase:replication to a separated file.

Done.

@2005hithlj PTAL.

Thanks.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 30s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ HBASE-27109/table_based_rqs Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  1s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  compile  |   3m  1s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  checkstyle  |   0m 47s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  spotless  |   0m 44s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m  0s |  HBASE-27109/table_based_rqs passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 10s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 51s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 48s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 48s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 42s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  14m 16s |  Patch does not cause any errors with Hadoop 3.2.4 3.3.4.  |
| +1 :green_heart: |  spotless  |   0m 42s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  spotbugs  |   2m 11s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 15s |  The patch does not generate ASF License warnings.  |
|  |   |  44m 46s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/5/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5157 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti spotless checkstyle compile |
| uname | Linux 6d3554003e64 5.4.0-1097-aws #105~18.04.1-Ubuntu SMP Mon Feb 13 17:50:57 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27109/table_based_rqs / f78fe5994a |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| Max. process+thread count | 79 (vs. ulimit of 30000) |
| modules | C: hbase-replication hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/5/console |
| versions | git=2.34.1 maven=3.8.6 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 53s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ HBASE-27109/table_based_rqs Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 13s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 21s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  compile  |   0m 59s |  HBASE-27109/table_based_rqs passed  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  HBASE-27109/table_based_rqs passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 19s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 59s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   4m 25s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 37s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 30s |  hbase-replication in the patch passed.  |
| -1 :x: |  unit  | 228m 57s |  hbase-server in the patch failed.  |
|  |   | 253m 50s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.42 ServerAPI=1.42 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/5/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/5157 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux c17ceca693c4 5.4.0-137-generic #154-Ubuntu SMP Thu Jan 5 17:03:22 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | HBASE-27109/table_based_rqs / f78fe5994a |
| Default Java | Eclipse Adoptium-11.0.17+8 |
| unit | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/5/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/5/testReport/ |
| Max. process+thread count | 2465 (vs. ulimit of 30000) |
| modules | C: hbase-replication hbase-server U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-5157/5/console |
| versions | git=2.34.1 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by 2005hithlj** [comment]
@Apache9 sir LTGM.
But some UTs failed, and it's unclear whether it's related to our code changes?

**Review by 2005hithlj — APPROVED** [review]

**Comment by Apache9** [comment]
Let me check the test logs for the failed UT.

**Comment by Apache9** [comment]
Posted the findings on HBASE-27277, TestRaceBetweenSCPAndTRSP is known to be flaky for a while, only in pre commit, anyway.

Let me merge this first.

We need to keep an eye on the nightly result for branch HBASE-27109, to fix the unstable tests.

Thanks.

**Pull request merged**